apps/web/src/common/modules/widgets/_widgets/table/Table.vue

```diff
@@ -32,66 +42,253 @@ const state = reactive({
+    tableDataMaxCount: computed<number>(() => state.tableDataFieldInfo?.count),
+    groupByField: computed<string[]|undefined>(() => ((props.widgetOptions?.groupBy as GroupByValue)?.value as string[]) ?? []),
     dateRange: computed<DateRange>(() => {
-        const _start = state.basedOnDate;
-        const _end = state.basedOnDate;
-        // if (state.xAxisField === DATE_FIELD) {
-        //     [_start, _end] = getWidgetDateRange(state.granularity, state.basedOnDate, state.xAxisCount);
-        // }
-        return { start: _start, end: _end };
+        let subtract = 1;
+        if (state.tableDataField === 'Date' || state.groupByField?.includes('Date')) {
+            if (state.granularity === GRANULARITY.YEARLY) subtract = 3;
+            if (state.granularity === GRANULARITY.MONTHLY) subtract = 12;
+            if (state.granularity === GRANULARITY.DAILY) subtract = 30;
+        }
+        const [start, end] = getWidgetDateRange(state.granularity, state.basedOnDate, subtract);
+        return { start, end };
+    }),
+    comparisonDateRange: computed<DateRange>(() => {
+        const [_start] = getWidgetDateRange(state.granularity, state.basedOnDate, 2);
+        return { start: _start, end: _start };
+    }),
+    // data for optional fields
+    isComparisonEnabled: computed<boolean>(() => state.tableDataField !== 'Date' && !state.groupByField?.includes('Date')),
+    comparisonInfo: computed<ComparisonValue|undefined>(() => props.widgetOptions?.comparison?.[0] as ComparisonValue),
+    subTotalInfo: computed<TotalValue|undefined>(() => props.widgetOptions?.subTotal as TotalValue),
+    totlaInfo: computed<TotalValue|undefined>(() => props.widgetOptions?.total as TotalValue),
+    progressBarInfo: computed<ProgressBarValue|undefined>(() => props.widgetOptions?.progressBar as ProgressBarValue),
+    // table
+    tableFields: computed<TableWidgetField[]>(() => {
+        const labelFields: TableWidgetField[] = state.groupByField?.map((field) => ({ name: field, label: field, fieldInfo: { type: 'labelField' } })) ?? [];
+        let dataFields: TableWidgetField[] = [];
+        if (state.tableDataFieldType === 'staticField') {
+            state.tableDataField.forEach((field) => {
+                dataFields.push({
+                    name: field,
+                    label: field,
+                    fieldInfo: { type: 'dataField' },
+                });
+                if (state.comparisonInfo?.format && state.isComparisonEnabled) {
+                    dataFields.push({
+                        name: `comparison_${field}`,
+                        label: 'Comparison',
+                        fieldInfo: { type: 'dataField', additionalType: 'comparison' },
+                    });
+                }
+            });
+        } else if (state.tableDataField === 'Date') dataFields = getWidgetTableDateFields(state.granularity, state.dateRange, state.tableDataMaxCount);
```

Comment:
nitpick: there's `DATE_FIELD` in common/modules/widgets/_constants/widget-contstant.ts